mcp-spring/mcp-spring-webflux/src/test/java/io/modelcontextprotocol/WebFluxSseIntegrationTests.java

```diff
@@ -189,7 +189,7 @@ void testCreateMessageSuccess(String clientType) {
 				assertThat(result.stopReason()).isEqualTo(CreateMessageResult.StopReason.STOP_SEQUENCE);
 			});
 		}
-		mcpServer.closeGracefully().block();
+		mcpServer.closeGracefully();
```

Comment:
This won't close the server since the Mono is not subscribed to. As this is an async server it is required to subscribe. I think the same issue has been introduced in multiple other places.

1. Please review all the code changes for this error.
2. Please do not force-push as it makes reviewing difficult.

Thanks!

Reply:
1. We should always return the same instance.
2. We should do the close/subscribe when invoking. 

But , these are not consistent across the codebase.

In akka/pekko, there is a `whenTerminate`, which will return the same instance too.

Reply:
wdyt about this @chemicL , I'm a little busy at work, but I can update this pr with your input.

Reply:
I'm not sure - is this in reference to my comment above? I only wanted to indicate that the `mcpServer` instance is an `McpAsyncServer`, which means `closeGracefully()` returns a `Mono`, which needs to be subscribed to in order to trigger closing, otherwise nothing happens.

That comment does not touch upon returning the same instance of `Mono`, just the fact that it won't close anything.

For returning the same instance - I am not sure whether we need that - do you think we do? I imagine in most cases the double closing should be idempotent so perhaps there is no need. If we do have issues with double closing, let's open a new issue to address this and think of solutions, WDYT?